designs/strings-design.md

```diff
@@ -0,0 +1,95 @@
+## TEST CASES FOR WEEK 7:
+* test case 1: var definition for string
+```
+s:str = "asdf"
+```
+
+* test case 2: print string
+```
+s:str = "asdf"
+print(s)
+```
+
+* test case 3: index accessing
+```
+s:str = "asdf"
+print(s[0])
+```
+
+* test case 4: index out of range error, should be a runtime error
+```
+s:str = "asdf"
+print(s[5])
+```
+
+* test case 5: get the length of string
+```
+s:str = "asdf"
+print(len(s))
+```
+
+* test case 6: immutable ???
+```
+s:str = "asdf"
+s[1] = "p"
```

Comment:
What does Python's semantics say should happen here? Are you planning on deviating from it?

Reply:
Python strings are immutable. They do not behave like a list of characters. Please refer to the updated design.md "Some Further nodes" section for a more detailed explaination.